src/rai_asr/rai_asr/asr_node.py

```diff
@@ -282,17 +286,20 @@ def capture_sound(self):
             )
             if asr_lock or self.hmi_lock or self.tts_lock:
                 continue
+                
 
-            self.audio_buffer.append(audio_data)
             if self.should_listen(audio_data):
                 self.silence_start_time = datetime.now()
                 if not self.is_recording:
                     self.start_recording()
                     self.reset_buffer()
+                    self.audio_buffer.append(audio_data)
             elif self.is_recording:
-                if datetime.now() - self.silence_start_time > self.grace_period:
+                self.audio_buffer.append(audio_data)
+                if datetime.now() - self.silence_start_time > timedelta(seconds=self.silence_grace_period):
                     self.stop_recording_and_transcribe()
                     self.reset_buffer()
+                    self.oww_model.reset()
```

Comment:
See the previous comment. I think we should contain the oww model logic in should_listen